### ISSUE TITLE
dsc_securityoption not idempotent

### ISSUE THREAD

**Issue by bwilcox** [puppetlabs/ruby-pwsh#82]
## Describe the Bug
dsc_securityoption resource applies corrective action with every puppet run.

## Expected Behavior
The resource should be idempotent and only change if needed.

## Steps to Reproduce
Resource example:

```
  $value = [
    'AES128_HMAC_SHA1',
    'FUTURE',
    'AES256_HMAC_SHA1',
  ]

  dsc_securityoption { 'Configure encryption types allowed for Kerberos':
    dsc_name                                                             => 'Configure encryption types allowed for Kerberos',
    dsc_network_security_configure_encryption_types_allowed_for_kerberos => $value
  }
```

## Environment
 - Version pwshlib 0.6.3, securitypolicydsc 2.10.0-0-1
 - Platform Windows 2019

## Additional Context

It looks like the resource state is not including the value of the parameter.

```
From: C:/Ruby25-x64/lib/ruby/gems/2.5.0/gems/puppet-resource_api-1.8.13/lib/puppet/resource_api.rb @ line 310 dsc_securityoption#flush:

    305:         # skip puppet's injected metaparams
    306:         actual_params = @parameters.select { |k, _v| type_definition.attributes.key? k }
    307:         target_state = Hash[actual_params.map { |k, v| [k, v.rs_value] }]
    308:         target_state = my_provider.canonicalize(context, [target_state]).first if type_definition.feature?('canonicalize')
    309:
 => 310:         retrieve unless @rsapi_current_state
    311:
    312:         return if @rsapi_current_state == target_state
    313:
    314:         Puppet.debug("Target State: #{target_state.inspect}")
    315:
[1] pry(#<Puppet::Type::Dsc_securityoption>)> @rsapi_current_state
=> {:name=>"Configure encryption types allowed for Kerberos",
 :dsc_name=>"Configure encryption types allowed for Kerberos"}
[2] pry(#<Puppet::Type::Dsc_securityoption>)> target_state
=> {:name=>"Configure encryption types allowed for Kerberos",
 :dsc_name=>"Configure encryption types allowed for Kerberos",
 :dsc_network_security_configure_encryption_types_allowed_for_kerberos=>
  ["AES128_HMAC_SHA1", "FUTURE", "AES256_HMAC_SHA1"]}
```

**Comment by michaeltlombardi** [comment]
Thanks for the detailed repro! We'll take a look - seems like the appropriate structure isn't being returned from the `Get` invocation, so the canonicalization is failing.

**Comment by michaeltlombardi** [comment]
This tickled my brain, this is actually a bug in the upstream module that I've already submitted a fix for (and had merged, but not released):

https://github.com/dsccommunity/SecurityPolicyDsc/pull/152

TLDR The DSC Resource does not return the `Name` property despite requiring it, so Puppet will always see that a corrective change is necessary.
